### ISSUE TITLE
Problem with Expo Print.printToFileAsync

### ISSUE THREAD

**Issue by JensSam** [expo/expo#6169]
### Environment

Expo CLI 3.4.1 environment info:
    System:
      OS: Windows 10
    Binaries:
      Yarn: 1.19.1 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 6.10.3 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  3.5.0.0 AI-191.8026.42.35.5900203

Target iOS 13.1.3

I've written a very short program which will print a PDF and share it as well as show the PDF in a modal. The HTML code includes two pictures: one that's online on a website, and one that is locally stored. In Android both pictures get printed into the PDF, as they should. In iOS only the online (http:// ) version gets printed. I've tried a million ways to get expo print to print local files but I have not succeded in iOS, only in Android. 

My code is:
 `import React, { Component } from 'react';

import {
  Text,
  TouchableHighlight,
  View,
  Image
} from 'react-native';

import * as Print from 'expo-print';
import * as FileSystem from 'expo-file-system';
import PDFReader from 'rn-pdf-reader-js';
import * as Sharing from 'expo-sharing';

export default class Example extends Component {
  constructor (props) {
    super(props);
    this.state = {
      pdfuri: 'https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf',
      visible: false,
      picUri: 'https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png',

    }
    this.setUri = this.setUri.bind(this);

  }

  async createPDF() {
    fil = '';

    FileSystem.downloadAsync(
      'https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png',
      FileSystem.documentDirectory + 'google.png'
    )
      .then(({ uri }) => {
        console.log('Finished downloading to ', uri);
        fil = uri;
        this.printa(uri);
      })
      .catch(error => {
        console.error(error);
      });

    
    
    
  }

  async printa (uri) {
    console.log('docs directory:', FileSystem.documentDirectory );

    const htmlcontent1 = '<html><body>Here is a picture from the web:<br><img src="https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png" width="300" height:"300" />';
    const htmlcontent2 = '<br><br>Here is the local version of the web picture:<br><img src="'+FileSystem.documentDirectory + 'google.png'+'" width="300" height:"300" /></body></br>';
    const htmlcontent = htmlcontent1 + htmlcontent2;
    let options = {
      html: htmlcontent,

    };
    
    try {
      file = await Print.printToFileAsync(options);
      Sharing.shareAsync(file.uri, {dialogTitle: 'Here is your PDF'})
      this.setState({pdfuri: file.uri})
    }
    catch(error) {console.error(error)}

  }

  
  render() {
    return(
      <View >
        <TouchableHighlight style={{marginTop: 30, width: '50%', height: 50, backgroundColor: 'blue'}} onPress={() => this.createPDF()}>
          <Text>Create PDF</Text>
        </TouchableHighlight>
        <TouchableHighlight style={{marginTop: 30, width: '50%', height: 50, backgroundColor: 'green'}} onPress={() => console.log(htmlcontent)}>
          <Text>Show htmlcontent1</Text>
        </TouchableHighlight>
        <TouchableHighlight style={{marginTop: 30}} onPress={() => this.setState({visible: !this.state.visible})}>
          <Text>Open Modal</Text>
        </TouchableHighlight>
        <Image style={{width: 200, height: 50}} source={{uri: FileSystem.documentDirectory + 'google.png'}} />
{this.state.visible ? 
        <View style={{marginTop: 20, backgroundColor: 'blue', borderWidth: 2, borderColor: 'red', margin: 5, height: 500}} >
          <PDFReader
            source={{
              uri: this.state.pdfuri,
            }}
          /> 
        </View>
        : null }
      </View>
    )
  }
}
`

I expect the iOS version to function as well as the Android version. I get absolutely no error messages, only an empty picture in the PDF. Since I've set the picture to a fixed size I can see there is space allocated for the picture, only that it is empty. 

And for your info when you try this example out, the PDF reader doesn't seem to work on iOS either, it just keeps on loading and loading. So I usually share the file to a drive I can read from my PC instead. It works well in the Android emulator though.

**Comment by cruzach** [comment]
Could you share a [snack](https://snack.expo.io/) that reproduces this? I tried using your code but it doesn't seem to run (looks like you're missing a few closing tags)

**Comment by EAT-CODE-KITE-REPEAT** [comment]
Here is a reproducible snack:

https://snack.expo.io/r1IBgIW2B

**Comment by teppa-rootstack** [comment]
hey guys did you find a solution I have the same issue when the img src comes from a local file uri.

**Comment by devaublanc** [comment]
Yes if you convert the local file in base64 before concat in your html template

```javascript
const manipResult = await ImageManipulator.manipulateAsync(localURI, [], { base64: true });
    const print = await Print.printToFileAsync({
      html: `
        <html>
         <img src="data:image/jpeg;base64,${manipResult.base64}" style="width:100%;" />
        </html>
      `,
    });
```
